### PR TITLE
Add ServiceConfiguration maxNumRetries and backoffTimeSlot

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: service-config/src/main/java/com/palantir/remoting/api/config/service/ServicesConfigBlock.java [hunk 1]
```diff
@@ -79,6 +79,12 @@
     public abstract Optional<HumanReadableDuration> defaultWriteTimeout();
 
     /**
+     * Default global write timeout.
```
Comment (bavardage) [inline service-config/src/main/java/com/palantir/remoting/api/config/service/ServicesConfigBlock.java]: this should probably match PartialServiceConfiguration.java#55

File: service-config/src/main/java/com/palantir/remoting/api/config/service/PartialServiceConfiguration.java [hunk 2]
```diff
@@ -49,6 +49,15 @@
     /** Write timeout for requests. */
     Optional<HumanReadableDuration> writeTimeout();
 
+    /** The maximum number of times a failed RPC call should be retried. */
+    Optional<Integer> maxNumRetries();
+
+    /**
+     * The size of one backoff time slot for call retries. For example, an exponential backoff retry algorithm may
```
Comment (bavardage) [inline service-config/src/main/java/com/palantir/remoting/api/config/service/PartialServiceConfiguration.java]: (not saying javadoc should change, just curious) and so a 'constant' backoff could just retry every `backoffSlotSize`, a `linear` backoff choose `c*backoffSlotSize`?

Comment (uschi2000) [reply service-config/src/main/java/com/palantir/remoting/api/config/service/PartialServiceConfiguration.java]: Yes

File: service-config/src/main/java/com/palantir/remoting/api/config/service/ServicesConfigBlock.java (hunk 1 above)
Comment (uschi2000) [reply service-config/src/main/java/com/palantir/remoting/api/config/service/ServicesConfigBlock.java]: done

**Pull request merged**